L1Trigger/TrackFindingTracklet/src/Tracklet.cc

```diff
@@ -644,149 +632,168 @@ void Tracklet::setFitPars(double rinvfit,
 }
 
-std::string Tracklet::trackfitstr() {
-  string stubid0 = "111111111";
-  string stubid1 = "111111111";
-  string stubid2 = "111111111";
-  string stubid3 = "111111111";
+const std::string Tracklet::layerstubstr(const unsigned layer) const {
+  assert(layer <= 5);
 
-  if (isBarrel()) {
-    if (layer() == 1) {
-      if (layerresid_[2].valid()) {
-        stubid0 = layerresid_[2].fpgastubid().str();
-      }
-      if (layerresid_[3].valid()) {
-        stubid1 = layerresid_[3].fpgastubid().str();
-      }
-      if (layerresid_[4].valid()) {
-        stubid2 = layerresid_[4].fpgastubid().str();
-      }
-      if (layerresid_[5].valid()) {
-        stubid3 = layerresid_[5].fpgastubid().str();
-      }
-      if (diskresid_[0].valid()) {
-        stubid3 = diskresid_[0].fpgastubid().str();
-      }
-      if (diskresid_[1].valid()) {
-        stubid2 = diskresid_[1].fpgastubid().str();
-      }
-      if (diskresid_[2].valid()) {
-        stubid1 = diskresid_[2].fpgastubid().str();
-      }
-      if (diskresid_[3].valid()) {
-        stubid0 = diskresid_[3].fpgastubid().str();
-      }
+  std::stringstream oss("");
+  if (!layerresid_[layer].valid())
+    oss << "0|0000000|0000000000|0000000|000000000000|000000000";
+  else {
+    if (trackIndex_ < 0 || trackIndex_ > (int)settings_.ntrackletmax()) {
+      cout << "trackIndex_ = " << trackIndex_ << endl;
+      assert(0);
     }
+    const FPGAWord tmp(trackIndex_, 7, true, __LINE__, __FILE__);
+    oss << "1|";  // valid bit
+    oss << tmp.str() << "|";
+    oss << layerresid_[layer].fpgastubid().str() << "|";
+    oss << layerresid_[layer].stubptr()->r().str() << "|";
+    oss << layerresid_[layer].fpgaphiresid().str() << "|";
+    oss << layerresid_[layer].fpgazresid().str();
+  }
 
-    if (layer() == 3) {
-      if (layerresid_[0].valid()) {
-        stubid0 = layerresid_[0].fpgastubid().str();
-      }
-      if (layerresid_[1].valid()) {
-        stubid1 = layerresid_[1].fpgastubid().str();
-      }
-      if (layerresid_[4].valid()) {
-        stubid2 = layerresid_[4].fpgastubid().str();
-      }
-      if (layerresid_[5].valid()) {
-        stubid3 = layerresid_[5].fpgastubid().str();
-      }
-      if (diskresid_[0].valid()) {
-        stubid3 = diskresid_[0].fpgastubid().str();
-      }
-      if (diskresid_[1].valid()) {
-        stubid2 = diskresid_[1].fpgastubid().str();
-      }
-    }
+  return oss.str();
+}
 
-    if (layer() == 5) {
-      if (layerresid_[0].valid()) {
-        stubid0 = layerresid_[0].fpgastubid().str();
-      }
-      if (layerresid_[1].valid()) {
-        stubid1 = layerresid_[1].fpgastubid().str();
-      }
-      if (layerresid_[2].valid()) {
-        stubid2 = layerresid_[2].fpgastubid().str();
-      }
-      if (layerresid_[3].valid()) {
-        stubid3 = layerresid_[3].fpgastubid().str();
-      }
-    }
-  }
+const std::string Tracklet::diskstubstr(const unsigned disk) const {
+  assert(disk <= 4);
```

Comment:
similarly here, can this be "disk < N_DISK" where N_DISK=5 from Settings.h?

Reply:
Done.